agent/tcs/handler/types_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package tcshandler
+
+import (
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsMetricsDisabled(t *testing.T) {
+	tcs := []struct {
+		param       *TelemetrySessionParams
+		result      bool
+		err         error
+		description string
+	}{
+		{
+			param:       &TelemetrySessionParams{},
+			result:      false,
+			err:         errors.New("Config is empty in the tcs session parameter"),
+			description: "Config not set should cause error",
+		},
+		{
+			param:       &TelemetrySessionParams{Cfg: &config.Config{DisableMetrics: false, DisableDockerHealthCheck: false}},
+			result:      false,
+			err:         nil,
+			description: "No metrics was disable should return false",
+		},
+		{
+			param:       &TelemetrySessionParams{Cfg: &config.Config{DisableMetrics: true, DisableDockerHealthCheck: false}},
+			result:      false,
+			err:         nil,
+			description: "Only telemetry metrics was disable should return false",
+		},
+		{
+			param:       &TelemetrySessionParams{Cfg: &config.Config{DisableMetrics: true, DisableDockerHealthCheck: true}},
+			result:      true,
+			err:         nil,
+			description: "both telemetry and health metrics were disable should return true",
+		},
```

Comment:
Is it in golang, for **&&** operator, if the first one is false, it will not proceed to check the value of second parameter? Since seems you miss a case, when **DisableMetrics** is false and **DisableDockerHealthCheck** is true.